### PR TITLE
Iteration 3

### PR DESCRIPTION
**Pull request by tarynmartin**
#### What’s this PR do?
- Clean up code based on linter feedback
- Add game initialization and called the game to start in index.js
- Refactor code in Round.js and Turn.js to fix a bug
- Refactor calculatePercentCorrect in Round.js to round decimals to a whole number
- Refactor code throughout to be cleaner and in-line with Javascript standards
#### Where should the reviewer start?
- Open Turn.js, Round.js, and index.js files in the src folder to see what's been created or refactored.
#### How should this be manually tested?
- Open terminal, clone this repo, run npm install, and then run node index.js to play the game!
#### Questions
- Code no longer passes calculatePercentCorrect method with the added toFixed() property? Not sure what to do there.

### CODE REVIEW

File: src/Round.js [hunk 1]
```diff
@@ -1,42 +1,45 @@
...
-            return turn.giveFeedback();
-        } else {
-            this.turns = this.turns + 1;
-            this.currentCard = this.cards[this.currentCard.id];
-            return turn.giveFeedback();
-        }
-    }
-    calculatePercentCorrect() {
-        return 100 - ((this.incorrectGuesses.length / this.turns) * 100);
+    switch (guessEval) {
```
Comment (hfaerber) [inline src/Round.js:18]: Nice refactoring to implement a switch statement.  Switch statements would be a great lightning talk topic.

File: src/Round.js [hunk 2]
```diff
@@ -1,42 +1,45 @@
...
+        return turn.giveFeedback();
+        break;
     }
-    endRound() {
-        if (this.currentCard === undefined) {
-            return `** Round over! ** You answered ${this.calculatePercentCorrect()}% of the questions correctly!`;
-        }
+  }
+  calculatePercentCorrect() {
+    const finalPercent = 100 - ((this.incorrectGuesses.length / this.turns) * 100);
```
Comment (hfaerber) [inline src/Round.js:32]: You could have wrapped your original return code in parens and tacked .toFixed() to the end of it but how you have it here is easier to read and therefore better.   Don't be tricked into thinking shorter code always means cleaner/better code.   Clear, readable code = better code, even if it is longer.  Good job!

File: src/Turn.js [hunk 3]
```diff
@@ -1,32 +1,34 @@
...
+  returnCard() {
+    return this.currentCard;
+  }
+  evaluateGuess() {
+    if (this.userGuess === this.currentCard.correctAnswer) {
+      this.guessCorrect = true;
+      return true;
+    } else {
+      this.guessCorrect = false;
+      return false;
```
Comment (hfaerber) [inline src/Turn.js]: There is a minor opportunity to refactor what/how you are returning what you need here.

File: src/Turn.js [hunk 4]
```diff
@@ -1,32 +1,34 @@
...
-    }
-    evaluateGuess() {
-        if (this.userGuess === this.currentCard.correctAnswer) {
-            this.guessCorrect = true;
-            return true;
-        } else {
-            this.guessCorrect = false;
-            return false;
-        }
-        //this.userGuess === this.currentCard.correctAnswer ? true : false;
```
Comment (hfaerber) [inline src/Turn.js:21]: Why did you decide against the ternary here? Its okay, just wondering what was behind that decision.
 My note about refactoring this evaluateGuess() method isn't asking for it to be refactored into a ternary necessarily - just to be clear.

File: src/Turn.js [hunk 5]
```diff
@@ -1,32 +1,34 @@
...
-        }
+  }
+  giveFeedback() {
+    switch(this.guessCorrect) {
+      case true:
+				return "correct!"
+				break;
+			case false:  
+				return "incorrect!";
+				break;
```
Comment (hfaerber) [inline src/Turn.js]: Weird spacing here.  What is the decision behind when to use if/else vs ternary vs switch/case?  Is it part of the spec, or for practice, or a more specific reason?

File: test/Deck-test.js [hunk 6]
```diff
@@ -6,31 +6,30 @@ const Deck = require('../src/Deck');
 
 describe('Deck', function () {
 
-    it('should be a function', function () {
-        const deck = new Deck();
-        expect(Deck).to.be.a('function');
-    });
+  it('should be a function', function () {
+    const deck = new Deck();
```
Comment (hfaerber) [inline test/Deck-test.js]: Do you need line 10 since your expect statement is only looking at "Deck" not "deck"?

File: test/Turn-test.js [hunk 7]
```diff
@@ -6,63 +6,63 @@ const Card = require('../src/Card');
 
 describe('Turn', function () {
 
-    it('should be a function', function () {
-        const turn = new Turn();
-        expect(Turn).to.be.a('function');
-    });
+  it('should be a function', function () {
+    const turn = new Turn();
```
Comment (hfaerber) [inline test/Turn-test.js]: Do you need line 10 here?

**Comment by hfaerber** [comment]
This looks great Taryn!  It is very clean and readable code.  Your testing looks through and accurate.  This mod you will learn about using 'beforeEach' in testing which would clean up some of your test setups here but I would not focus on that for this project unless you are done with literally everything else and looking for something to do.  Great work!

File: src/Turn.js (hunk 4 above)
Comment (tarynmartin) [reply src/Turn.js:21]: I think I originally tried this and it wasn't passing the test. I'll play with this method and see what I can do to refactor.

File: src/Turn.js (hunk 5 above)
Comment (tarynmartin) [reply src/Turn.js]: Switch is more practice than anything. Trying to practice using ternaries to see where and when I can use them. Clean code is pretty.

**Pull request merged**